### PR TITLE
Haiku build fixes

### PR DESCRIPTION
**Pull request by extrowerk**
Some small build fixes for Haiku

### CODE REVIEW

File: src/iohandler/process_io_handler.cc [hunk 1]
```diff
@@ -40,6 +40,10 @@
 #include "content/content_manager.h"
 #include "util/process.h"
 
+#ifndef FD_SET
+#include <sys/select.h>
```
Comment (neheb) [inline src/iohandler/process_io_handler.cc:44]: Move it up before sys/stat and remove the ifndef

File: src/util/tools.cc [hunk 2]
```diff
@@ -48,6 +48,10 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+#ifdef __HAIKU__
+#define _DEFAULT_SOURCE
```
Comment (neheb) [inline src/util/tools.cc:52]: This is interesting. Why is DEFAULT_SOURCE needed?

**Comment by neheb** [comment]
I am somewhat curious why sys/select.h is being included on every other platform. No libc++ header includes it, meaning it's some libc header implicitly including it.

**Comment by neheb** [comment]
I see. Lazy libc++ coding. sys/time.h (which includes sys/select.h) gets unconditionally included in the filesystem header while it's only conditionally used.

File: src/util/tools.cc (hunk 2 above)
Comment (extrowerk) [reply src/util/tools.cc:52]: It guards the BSD additions.
https://github.com/haiku/haiku/blob/master/headers/compatibility/bsd/ifaddrs.h#L10

Comment (neheb) [reply src/util/tools.cc:52]: Huh. Interesting.

Comment (extrowerk) [reply src/util/tools.cc:52]: Maybe just features.h should be included, idk.
https://github.com/haiku/haiku/blob/master/headers/compatibility/bsd/features.h

**Comment by whyman** [comment]
Thanks!

Would be good to get a doc for the install page to add Haiku :+1:

**Pull request merged**